aicsimageio/readers/bioformats_reader.py

```diff
@@ -0,0 +1,442 @@
+
+    @cached_property
+    def ome_metadata(self) -> OME:
+        """Return OME object parsed by ome_types."""
+        with LociFile(self._path) as rdr:
+            meta = rdr.ome_metadata
+        return meta
+
+    @property
+    def physical_pixel_sizes(self) -> types.PhysicalPixelSizes:
+        """
+        Returns
+        -------
+        sizes: PhysicalPixelSizes
+            Using available metadata, the floats representing physical pixel sizes for
+            dimensions Z, Y, and X.
+
+        Notes
+        -----
+        We currently do not handle unit attachment to these values. Please see the file
+        metadata for unit information.
+        """
+        return metadata_utils.physical_pixel_sizes(
+            self.metadata, self.current_scene_index
+        )
+
+    def _to_xarray(self, delayed: bool = True) -> xr.DataArray:
+        with LociFile(self._path) as rdr:
+            image_data = rdr.to_dask() if delayed else rdr.to_numpy()
+            _, coords = metadata_utils.get_dims_and_coords_from_ome(
+                ome=rdr.ome_metadata,
+                scene_index=self.current_scene_index,
+            )
+
+        return xr.DataArray(
+            image_data,
+            dims=dimensions.DEFAULT_DIMENSION_ORDER_LIST_WITH_SAMPLES
+            if rdr.core_meta.is_rgb
+            else dimensions.DEFAULT_DIMENSION_ORDER_LIST,
+            coords=coords,  # type: ignore
+            attrs={
+                constants.METADATA_UNPROCESSED: rdr.ome_xml,
+                constants.METADATA_PROCESSED: rdr.ome_metadata,
+            },
+        )
+
+    @staticmethod
+    def bioformats_version() -> str:
+        """The version of the bioformats_package.jar being used."""
+        from bioformats_jar import get_loci
+
+        return get_loci().__version__
+
+
+class CoreMeta(NamedTuple):
+    """NamedTuple with core bioformats metadata. (not OME meta)"""
+
+    shape: Tuple[int, int, int, int, int, int]
+    dtype: np.dtype
+    series_count: int
+    is_rgb: bool
+    is_interleaved: bool
+    dimension_order: str
+    resolution_count: int
+
+
+class LociFile:
+    """Read image and metadata from file supported by Bioformats.
+
+    LociFile instances must be closed using the 'close' method, which is
+    automatically called when using the 'with' context manager.
+
+    LociFile instances are not thread-safe.
+
+    Parameters
+    ----------
+    path : str or Path
+        path to file
+    series : int, optional
+        the image series to read, by default 0
+    meta : bool, optional
+        whether to get metadata as well, by default True
+    memoize : int, optional
+        threshold for memoizing the reader. If the time required to call
+        `reader.setId()` is larger than this number, the initialized reader
+        (including all reader wrappers) will be cached in a memo file for later
+        retrieval. by default 50.  set to 0 or less to turn off memoization.
```

Comment:
what are the units of this number?

Reply:
milliseconds.  adding a comment.
Actually, I want to generally draw your attention here.  This is a nice feature, but I think it has to be off by default: it writes a memo file to the same directory as the file being opened (or the location of the BIOFORMATS_MEMO_DIR if set).  While it will reduce the time to load the file on future loads, it can actually be a couple megabytes depending on the size of the original file.  So, just wanted you to notice it's there (particularly if we find that we need to go back to a dask mode where the file is opened and closed for every chunk)... but it's off for now

Reply:
I agree it should be off by default. (We could add a reader kwarg to forward to this if it makes sense for users to want to control it) What does it do if you don't have write permission?  Is it an error or does it just silently not memoize?

Reply:
> Is it an error or does it just silently not memoize?

it's a good question!  I suspect it errors, but I haven been able to confirm.  something for a future PR